internal/cache/controller_obj_cache.go

```diff
@@ -2700,27 +2700,15 @@ func checkAndSetCloudType(client *clients.AviClient) bool {
 func checkPublicCloud(client *clients.AviClient) bool {
 	if lib.IsPublicCloud() {
 		// Handle all public cloud validations here
-		networkName := lib.GetNetworkName()
-		if networkName == "" && lib.GetCloudType() != lib.CLOUD_GCP {
-			if lib.GetCloudType() == lib.CLOUD_AWS {
-				//check for vipNetworkList if networkName not present for AWS Clouds
-				utils.AviLog.Infof("networkName not specified, checking for vipNetworkList as cloud is AWS")
-				vipNetworkList, err := lib.GetVipNetworkList()
-				if err == nil && len(vipNetworkList) != 0 {
-					utils.AviLog.Infof("Proceeding as multi-vip enabled")
-					return true
-				} else {
-					utils.AviLog.Errorf("vipNetworkList not specified, syncing will be disabled.")
-					if err != nil {
-						utils.AviLog.Errorf("%s", err.Error())
-					}
-					return false
-				}
-			}
-			// networkName is required param for Azure Clouds
-			utils.AviLog.Errorf("Required param networkName not specified, syncing will be disabled.")
-			return false
+		vipNetworkList, err := lib.GetVipNetworkList()
+		if err == nil && len(vipNetworkList) != 0 {
+			return true
 		}
+		utils.AviLog.Errorf("vipNetworkList not specified, syncing will be disabled.")
+		if err != nil {
+			utils.AviLog.Errorf("Got error while fetching VIP network list: %s", err.Error())
+		}
```

Comment:
The error message at 2707 seems misleading if there was an error in getting the vip network list. We will print both the error messages in that case. Can we check for the specific conditions for both the cases:
1. vip network list is not specified.
2. error is not nil while getting vip network list.